### PR TITLE
Add note listing page UI, necessary components and neetoUI V2

### LINKED ISSUE #1: Create note listing page UI

### PR DESCRIPTION
**Pull request by sahilLikes2Code**
Closes #1

### CODE REVIEW

File: app/javascript/src/components/Common/Navbar/index.jsx [hunk 1]
```diff
@@ -23,30 +22,49 @@ const NavBar = () => {
...
+            icon: UserCircle,
+            label: "Users",
+            to: "/users",
+          },
+          {
+            icon: NeetoInsights,
+            label: "Reports",
+            to: "#",
+          },
+        ]}
```
Comment (karthiknmenon) [inline app/javascript/src/components/Common/Navbar/index.jsx:52]: Please move this to a constants file.

File: app/javascript/src/components/Dashboard/Notes/Card.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          <Avatar
+            user={{
+              name: "Kieran Miller",
+              imageUrl: "https://randomuser.me/api/portraits/women/10.jpg",
+            }}
+            size="small"
+          />
+        </div>
+      </div>
+    </div>
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Notes/Card.jsx:39]: Let's split this further to be as follows - `Header, Body, Footer`.

File: app/javascript/src/components/Dashboard/Notes/Main.jsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import EmptyState from "components/Common/EmptyState";
+
+import DeleteAlert from "./DeleteAlert";
+import NewNotePane from "./NewNotePane";
+import NotesList from "./NotesList";
+
+import dummyNotes from "../../../constants/dummyNotes";
+
+const Main = () => {
+  const [loading, setLoading] = useState(true);
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Notes/Main.jsx:18]:
```js
const [isLoading, setIsLoading] = useState(true);
```

File: app/javascript/src/components/Dashboard/Notes/Main.jsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    <>
+      <Header
+        title="All Notes"
+        actionBlock={
+          <>
+            <Input
+              prefix={<Search />}
+              className="mr-3"
+              placeholder="Search Name, Email, Phone Number, Etc."
+              style={{ width: "305px" }}
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Notes/Main.jsx:56]: Please refrain from using inline styles. Use tailwind classnames.

File: app/javascript/src/components/Dashboard/Notes/Main.jsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            />
+            {/*increase the size of Input as per wireframe*/}
+            <Button
+              onClick={() => setShowNewNotePane(true)}
+              label="Add Note"
+              icon="ri-add-line"
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Notes/Main.jsx:62]: Please use neetoIcons here.

File: app/javascript/src/constants/dummyNotes.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+const dummyNotes = [
```
Comment (karthiknmenon) [inline app/javascript/src/constants/dummyNotes.js:1]: Please move constants file to the scope of the feature. This constants file can be placed inside the `Notes` directory. 
Let's extract out constants to a global file iff it's being reused at multiple places.

Comment (karthiknmenon) [reply app/javascript/src/constants/dummyNotes.js:1]:
```js
export const DUMM_NOTES
```

File: app/javascript/src/components/Dashboard/Notes/index.jsx [hunk 7]
```diff
@@ -1,96 +1,14 @@
...
-        fetchNotes={fetchNotes}
-      />
-      {showDeleteAlert && (
-        <DeleteAlert
-          selectedNoteIds={selectedNoteIds}
-          onClose={() => setShowDeleteAlert(false)}
-          refetch={fetchNotes}
-        />
-      )}
+      <UnequalTwoColumnSplitter menu={<SideMenu />} mainContent={<Main />} />
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Notes/index.jsx:11]: Could you please explain this further?

File: app/javascript/src/components/Common/Navbar/index.jsx [hunk 8]
```diff
@@ -1,15 +1,14 @@
 import React from "react";
 
-import { Toastr } from "neetoui";
-import { withRouter } from "react-router-dom";
+import { UserCircle, Text, NeetoInsights } from "@bigbinary/neeto-icons";
+import { Toastr } from "@bigbinary/neetoui/v2";
+import { Sidebar } from "@bigbinary/neetoui/v2/layouts";
```
Comment (karthiknmenon) [inline app/javascript/src/components/Common/Navbar/index.jsx:5]: Please create aliases.

**Review by karthiknmenon — COMMENTED** [review]
@sahilLikes2Code Please check the suggestions.

File: app/javascript/src/components/Dashboard/Notes/index.jsx (hunk 7 above)
Comment (sahilLikes2Code) [reply app/javascript/src/components/Dashboard/Notes/index.jsx:11]: Yes. Its purpose is to act as a layout component which can be fed two different components and their width, height, etc will be set to defined values.

The approach is prevent repetition of these settings across pages Notes and Contact that have similar layout a "Sidemenu" on the left and "Main content" on the right.

File: app/javascript/src/constants/dummyNotes.js (hunk 6 above)
Comment (sahilLikes2Code) [reply app/javascript/src/constants/dummyNotes.js:1]: Got it.

**Pull request merged**